src/Sarif/Core/FileData.cs

```diff
@@ -42,9 +66,18 @@ public static FileData Create(Uri uri, bool computeHashes)
                                 Algorithm = AlgorithmKind.Sha256,
                             },
                         };
+                }
+
             }
+            catch (Exception e) when (e is IOException || e is UnauthorizedAccessException) { }
```

Comment:
>UnauthorizedAccessException [](start = 63, length = 27)

Someone tell me how to mock File.ReadAllText to raise this exception. It's hard to contrive this case from .NET, need an AppDomain that both creates a file, then cleans it up later.

Reply:
As we discussed: See `MakeMockFileSystem`

---
In reply to: [136142197](https://github.com/Microsoft/sarif-sdk/pull/612#discussion_r136142197) [](ancestors = 136142197)